src/Data/Nat/Binary/Base.agda

```diff
@@ -106,6 +113,38 @@ _        * zero     =  zero
 1+[2 x ] * 2[1+ y ] =  2[1+ y + x * 2[1+ y ] ]
 1+[2 x ] * 1+[2 y ] =  1+[2 x + y * 1+[2 x ] ]
 
+_-_ : ℕᵇ → ℕᵇ → Maybe ℕᵇ
+x        - zero     = just x
+zero     - _        = nothing
+2[1+ x ] - 2[1+ y ] = map double (x - y)
+2[1+ x ] - 1+[2 y ] = map 1+[2_] (x - y)
+1+[2 x ] - 2[1+ y ] = map 1+[2_] (x - y >>= predᴹ)
+1+[2 x ] - 1+[2 y ] = map double (x - y)
+
+_divMod_ : ℕᵇ → ℕᵇ → ℕᵇ × ℕᵇ
+m divMod n = maybe′ (flip′ helper (pred n)) (zero , zero) (predᴹ m)
+  module divMod where
+  -- helper m n computes (suc m / suc n, suc m % suc n)
+  helper : ℕᵇ → ℕᵇ → ℕᵇ × ℕᵇ
+  helper zero     zero = 1+[2 zero ] , zero
+  helper zero     _    = zero        , 1+[2 zero ]
+  helper 2[1+ m ] n    = let q , r = helper m n
+                             q₂    = double q
+                             r₂    = double r
+                         in maybe′ (1+[2 q ] ,_) (q₂ , 1+[2 r ]) (r₂ - n)
+  helper 1+[2 m ] n    = let q , r = helper m n
+                             q₂    = double q
+                             r₂    = double r
+                         in maybe′ (1+[2 q ] ,_) (q₂ , r₂) (r₂ - n >>= predᴹ)
+
+_div_ = proj₁ ∘₂′ _divMod_
```

Comment:
We should follow the same conventions as in `Data.Nat.DivMod` and should reserve the names `_div_` and `_mod_` and indeed `_divMod_` operators for the variants that return `Fin`. I suggest removing `_div_` and `_mod_` entirely and rename `_divMod_` to `_/%_`?